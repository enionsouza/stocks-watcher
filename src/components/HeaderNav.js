import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import { BsChevronLeft } from 'react-icons/bs';
import { FaMicrophone } from 'react-icons/fa';
import { IoIosSettings } from 'react-icons/io';

const HeaderNav = ({ type, symbol }) => {
  const title = {
    home: 'Stocks Watcher',
    details: 'Details',
    noMatch: 'Error',
  };

  return (
    <div className="d-flex justify-content-around align-items-center py-1 dark-bg position-sticky">
      <Link to="/" className="link-button mx-3 fs-5">
        {type !== 'home' ? <BsChevronLeft /> : ''}
      </Link>
      <h3 className="body-title-text flex-grow-1 text-center mb-0 pt-1">{`${title[type]}${symbol ? ` - ${symbol}` : ''}`}</h3>
      <button
        type="button"
        onClick={() => {}}
        className="link-button fs-5"
      >
        <FaMicrophone />
      </button>
      <button
        type="button"
        onClick={() => {}}
        className="link-button mx-3 fs-5"
      >
        <IoIosSettings />
      </button>
    </div>
  );
};

HeaderNav.propTypes = {
  type: PropTypes.string.isRequired,
  // eslint-disable-next-line react/require-default-props
  symbol: PropTypes.string,
};

HeaderNav.default = {
  type: 'home',
};

export default HeaderNav;
