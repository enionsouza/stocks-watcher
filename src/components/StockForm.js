import { useState, useEffect } from 'react';
import { bindActionCreators } from 'redux';
import { useDispatch } from 'react-redux';
import { PropTypes } from 'prop-types';
import Form from 'react-bootstrap/Form';
import { filterStocks } from '../redux/Home/stocksList';

const StockForm = ({ allStocks }) => {
  const [stockSearch, setStockSearch] = useState('');
  const [exchangeSearch, setExchangeSearch] = useState('ALL');
  const [typeSearch, setTypeSearch] = useState('ALL');

  const dispatch = useDispatch();
  const filterStocksAction = bindActionCreators(filterStocks, dispatch);

  useEffect(() => {
    filterStocksAction(allStocks, stockSearch, exchangeSearch, typeSearch);
    return null;
  }, [stockSearch, exchangeSearch, typeSearch]);

  return (
    <Form className="d-flex my-3 control-text position-sticky">
      <Form.Control size="sm" type="text" placeholder="Stock search..." value={stockSearch} onChange={(e) => setStockSearch(e.target.value)} />
      <Form.Select value={exchangeSearch} onChange={(e) => setExchangeSearch(e.target.value)}>
        <option value="ALL">All Exchanges</option>
        <option value="AMEX">New York Stock Exchange Arca</option>
        <option value="NASDAQ">Nasdaq Global Market</option>
        <option value="NYSE">New York Stock Exchange</option>
        <option value="EURONEXT">EURONEXT</option>
        <option value="TSX">TSX (Toronto)</option>
        <option value="XETRA">XETRA (Germany)</option>
        <option value="LSE">LSE (London)</option>
        <option value="SIX">SIX (Zurich)</option>
        <option value="JNB">JNB (Johannesburg)</option>
        <option value="MCX">MCX (India)</option>
        <option value="NSE">NSE (India)</option>
        <option value="HKSE">HKSE (Hong Kong)</option>
        <option value="OSE">OSE (Osaka)</option>
        <option value="JPX">JPX (Tokio)</option>
      </Form.Select>
      <Form.Select value={typeSearch} onChange={(e) => setTypeSearch(e.target.value)}>
        <option value="ALL">All types</option>
        <option value="stock">Stocks</option>
        <option value="etf">ETFs</option>
        <option value="fund">Funds</option>
      </Form.Select>
    </Form>
  );
};

StockForm.propTypes = {
  allStocks: PropTypes.arrayOf(PropTypes.shape({
    exchange: PropTypes.string,
    exchangeShortName: PropTypes.string,
    name: PropTypes.string,
    symbol: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    tradable: PropTypes.bool.isRequired,
  })).isRequired,
};

export default StockForm;
