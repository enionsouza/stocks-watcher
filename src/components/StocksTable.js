import { Link } from 'react-router-dom';
import { useState, useEffect } from 'react';
import { PropTypes } from 'prop-types';
import Card from 'react-bootstrap/Card';
import Pagination from 'react-bootstrap/Pagination';

const StocksTable = ({ filteredStocks }) => {
  const cardsPerPage = 12;
  const lastPage = Math.ceil(filteredStocks.length / cardsPerPage);
  const [activePage, setActivePage] = useState(1);

  useEffect(() => {
    setActivePage(1);
    return null;
  }, [filteredStocks]);

  const renderTable = () => {
    const result = [];
    for (
      let i = (activePage - 1) * cardsPerPage;
      i < Math.min(cardsPerPage * activePage, filteredStocks.length);
      i += 2) {
      result.push(
        <div className="d-flex" key={filteredStocks[i + 1] ? `${filteredStocks[i].symbol}-${filteredStocks[i + 1].symbol}` : `${filteredStocks[i].symbol}`}>
          <Card className={`${i % 4 === 0 ? 'medium-bg' : 'dark-bg'} rounded-0`} style={{ width: '50%' }}>
            <Card.Body>
              <Card.Title>
                {filteredStocks[i].symbol}
                {' '}
                -
                {' '}
                {filteredStocks[i].price}
              </Card.Title>
              <Card.Subtitle className="mb-2">
                {filteredStocks[i].exchange}
                {' '}
                -
                {' '}
                {filteredStocks[i].exchangeShortName}
              </Card.Subtitle>
              <Card.Text>
                {filteredStocks[i].name}
                {' '}
                -
                {' '}
                {filteredStocks[i].type}
              </Card.Text>
              <Link to={`/details/${filteredStocks[i].symbol}`} className="link-button mx-3 fs-5">
                See details...
              </Link>
            </Card.Body>
          </Card>
          {filteredStocks[i + 1]
          && (
          <Card className={`${i % 4 === 0 ? 'dark-bg' : 'medium-bg'} rounded-0`} style={{ width: '50%' }}>
            <Card.Body>
              <Card.Title>
                {filteredStocks[i + 1].symbol}
                {' '}
                -
                {' '}
                {filteredStocks[i + 1].price}
              </Card.Title>
              <Card.Subtitle className="mb-2">
                {filteredStocks[i + 1].exchange}
                {' '}
                -
                {' '}
                {filteredStocks[i + 1].exchangeShortName}
              </Card.Subtitle>
              <Card.Text>
                {filteredStocks[i + 1].name}
                {' '}
                -
                {' '}
                {filteredStocks[i + 1].type}
              </Card.Text>
              <Link to={`/details/${filteredStocks[i + 1].symbol}`} className="link-button mx-3 fs-5">
                See details...
              </Link>
            </Card.Body>
          </Card>
          )}
        </div>,
      );
    }
    return result;
  };

  const pagination = () => {
    const items = [];
    const item = (i) => (
      <Pagination.Item key={`page-${i}`} active={activePage === i} onClick={() => setActivePage(i)}>
        {i}
      </Pagination.Item>
    );

    if (lastPage < 8) {
      for (let i = 1; i <= lastPage; i += 1) {
        items.push(item(i));
      }
    } else {
      items.push(item(1));
      items.push(<Pagination.Ellipsis key="ellipsis-1" />);
      if (activePage === 1 || activePage === 2) {
        for (let i = 2; i <= 4; i += 1) {
          items.push(item(i));
        }
      } else if (activePage === lastPage || activePage === lastPage - 1) {
        for (let i = lastPage - 3; i <= lastPage - 1; i += 1) {
          items.push(item(i));
        }
      } else {
        for (let i = activePage - 1; i <= activePage + 1; i += 1) {
          items.push(item(i));
        }
      }
      items.push(<Pagination.Ellipsis key="ellipsis-2" />);
      items.push(item(lastPage));
    }

    return items;
  };

  return (
    <>
      <Pagination className="d-flex justify-content-center" size="sm">
        {pagination()}
      </Pagination>
      {renderTable()}
    </>
  );
};

StocksTable.propTypes = {
  filteredStocks: PropTypes.arrayOf(PropTypes.shape({
    exchange: PropTypes.string,
    exchangeShortName: PropTypes.string,
    name: PropTypes.string,
    symbol: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    tradable: PropTypes.bool.isRequired,
  })).isRequired,
};

export default StocksTable;
