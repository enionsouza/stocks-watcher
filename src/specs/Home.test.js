import { Provider } from 'react-redux';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../redux/configureStore';
import HeaderNav from '../components/HeaderNav';
import StockForm from '../components/StockForm';
import StocksTable from '../components/StocksTable';
import StocksTablePlaceholder from '../components/StocksTablePlaceholder';
import allStocksMock from './mockData/allStocksMock';

describe('Home page', () => {
  jest.setTimeout(300000);
  describe('Use Jest snapshots to test the Home page', () => {
    it('renders Home page placeholders as expected', async () => {
      const homePage = render(
        <Provider store={store}>
          <Router>
            <Switch>
              <Route exact path="/">
                <HeaderNav type="home" />
                <StockForm allStocks={allStocksMock} />
                <StocksTablePlaceholder />
              </Route>
            </Switch>
          </Router>
        </Provider>,
      );

      expect(homePage.asFragment()).toMatchSnapshot();
    });
    it('renders Home page stocks table as expected', async () => {
      const homePage = render(
        <Provider store={store}>
          <Router>
            <Switch>
              <Route exact path="/">
                <HeaderNav type="home" />
                <StockForm allStocks={allStocksMock} />
                <StocksTable filteredStocks={allStocksMock} />
              </Route>
            </Switch>
          </Router>
        </Provider>,
      );

      expect(homePage.asFragment()).toMatchSnapshot();
    });
  });
});
