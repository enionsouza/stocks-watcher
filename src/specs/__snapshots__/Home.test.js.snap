// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page Use Jest snapshots to test the Home page renders Home page placeholders as expected 1`] = `
<DocumentFragment>
  <div
    class="d-flex justify-content-around align-items-center py-1 dark-bg position-sticky"
  >
    <a
      class="link-button mx-3 fs-5"
      href="/"
    />
    <h3
      class="body-title-text flex-grow-1 text-center mb-0 pt-1"
    >
      Stocks Watcher
    </h3>
    <button
      class="link-button fs-5"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 640 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M633.82 458.1l-157.8-121.96C488.61 312.13 496 285.01 496 256v-48c0-8.84-7.16-16-16-16h-16c-8.84 0-16 7.16-16 16v48c0 17.92-3.96 34.8-10.72 50.2l-26.55-20.52c3.1-9.4 5.28-19.22 5.28-29.67V96c0-53.02-42.98-96-96-96s-96 42.98-96 96v45.36L45.47 3.37C38.49-2.05 28.43-.8 23.01 6.18L3.37 31.45C-2.05 38.42-.8 48.47 6.18 53.9l588.36 454.73c6.98 5.43 17.03 4.17 22.46-2.81l19.64-25.27c5.41-6.97 4.16-17.02-2.82-22.45zM400 464h-56v-33.77c11.66-1.6 22.85-4.54 33.67-8.31l-50.11-38.73c-6.71.4-13.41.87-20.35.2-55.85-5.45-98.74-48.63-111.18-101.85L144 241.31v6.85c0 89.64 63.97 169.55 152 181.69V464h-56c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16z"
        />
      </svg>
    </button>
    <button
      class="link-button mx-3 fs-5"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 6a3.939 3.939 0 0 0-3.934 3.934h2C10.066 8.867 10.934 8 12 8s1.934.867 1.934 1.934c0 .598-.481 1.032-1.216 1.626a9.208 9.208 0 0 0-.691.599c-.998.997-1.027 2.056-1.027 2.174V15h2l-.001-.633c.001-.016.033-.386.441-.793.15-.15.339-.3.535-.458.779-.631 1.958-1.584 1.958-3.182A3.937 3.937 0 0 0 12 6zm-1 10h2v2h-2z"
        />
        <path
          d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
        />
      </svg>
    </button>
  </div>
  <form
    class="d-flex my-3 control-text position-sticky"
  >
    <input
      class="form-control form-control-sm"
      placeholder="Stock search..."
      type="text"
      value=""
    />
    <select
      class="form-select"
    >
      <option
        value="ALL"
      >
        All Exchanges
      </option>
      <option
        value="AMEX"
      >
        New York Stock Exchange Arca
      </option>
      <option
        value="NASDAQ"
      >
        Nasdaq Global Market
      </option>
      <option
        value="NYSE"
      >
        New York Stock Exchange
      </option>
    </select>
    <select
      class="form-select"
    >
      <option
        value="ALL"
      >
        All types
      </option>
      <option
        value="stock"
      >
        Stocks
      </option>
      <option
        value="etf"
      >
        ETFs
      </option>
      <option
        value="fund"
      >
        Funds
      </option>
    </select>
  </form>
  <ul
    class="d-flex justify-content-center pagination pagination-sm"
  >
    <div
      class="col-1 placeholder card-title h5"
    />
    <div
      class="col-1 placeholder card-title h5"
    />
    <div
      class="col-1 placeholder card-title h5"
    />
    <div
      class="col-1 placeholder card-title h5"
    />
    <div
      class="col-1 placeholder card-title h5"
    />
  </ul>
  <div
    class="d-flex"
  >
    <div
      class="medium-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="placeholder-glow card-title h5"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <div
          class="mb-2 placeholder-glow card-subtitle h6"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <p
          class="placeholder-glow card-text"
        >
          <span
            class="col-6 placeholder"
          />
        </p>
        <button
          class="link-button mx-3 fs-5 col-6 placeholder btn btn-light"
          disabled=""
          tabindex="-1"
          type="button"
        />
      </div>
    </div>
    <div
      class="dark-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="placeholder-glow card-title h5"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <div
          class="mb-2 placeholder-glow card-subtitle h6"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <p
          class="placeholder-glow card-text"
        >
          <span
            class="col-6 placeholder"
          />
        </p>
        <button
          class="link-button mx-3 fs-5 col-6 placeholder btn btn-light"
          disabled=""
          tabindex="-1"
          type="button"
        />
      </div>
    </div>
  </div>
  <div
    class="d-flex"
  >
    <div
      class="dark-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="placeholder-glow card-title h5"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <div
          class="mb-2 placeholder-glow card-subtitle h6"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <p
          class="placeholder-glow card-text"
        >
          <span
            class="col-6 placeholder"
          />
        </p>
        <button
          class="link-button mx-3 fs-5 col-6 placeholder btn btn-light"
          disabled=""
          tabindex="-1"
          type="button"
        />
      </div>
    </div>
    <div
      class="medium-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="placeholder-glow card-title h5"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <div
          class="mb-2 placeholder-glow card-subtitle h6"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <p
          class="placeholder-glow card-text"
        >
          <span
            class="col-6 placeholder"
          />
        </p>
        <button
          class="link-button mx-3 fs-5 col-6 placeholder btn btn-light"
          disabled=""
          tabindex="-1"
          type="button"
        />
      </div>
    </div>
  </div>
  <div
    class="d-flex"
  >
    <div
      class="medium-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="placeholder-glow card-title h5"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <div
          class="mb-2 placeholder-glow card-subtitle h6"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <p
          class="placeholder-glow card-text"
        >
          <span
            class="col-6 placeholder"
          />
        </p>
        <button
          class="link-button mx-3 fs-5 col-6 placeholder btn btn-light"
          disabled=""
          tabindex="-1"
          type="button"
        />
      </div>
    </div>
    <div
      class="dark-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="placeholder-glow card-title h5"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <div
          class="mb-2 placeholder-glow card-subtitle h6"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <p
          class="placeholder-glow card-text"
        >
          <span
            class="col-6 placeholder"
          />
        </p>
        <button
          class="link-button mx-3 fs-5 col-6 placeholder btn btn-light"
          disabled=""
          tabindex="-1"
          type="button"
        />
      </div>
    </div>
  </div>
  <div
    class="d-flex"
  >
    <div
      class="dark-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="placeholder-glow card-title h5"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <div
          class="mb-2 placeholder-glow card-subtitle h6"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <p
          class="placeholder-glow card-text"
        >
          <span
            class="col-6 placeholder"
          />
        </p>
        <button
          class="link-button mx-3 fs-5 col-6 placeholder btn btn-light"
          disabled=""
          tabindex="-1"
          type="button"
        />
      </div>
    </div>
    <div
      class="medium-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="placeholder-glow card-title h5"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <div
          class="mb-2 placeholder-glow card-subtitle h6"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <p
          class="placeholder-glow card-text"
        >
          <span
            class="col-6 placeholder"
          />
        </p>
        <button
          class="link-button mx-3 fs-5 col-6 placeholder btn btn-light"
          disabled=""
          tabindex="-1"
          type="button"
        />
      </div>
    </div>
  </div>
  <div
    class="d-flex"
  >
    <div
      class="medium-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="placeholder-glow card-title h5"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <div
          class="mb-2 placeholder-glow card-subtitle h6"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <p
          class="placeholder-glow card-text"
        >
          <span
            class="col-6 placeholder"
          />
        </p>
        <button
          class="link-button mx-3 fs-5 col-6 placeholder btn btn-light"
          disabled=""
          tabindex="-1"
          type="button"
        />
      </div>
    </div>
    <div
      class="dark-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="placeholder-glow card-title h5"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <div
          class="mb-2 placeholder-glow card-subtitle h6"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <p
          class="placeholder-glow card-text"
        >
          <span
            class="col-6 placeholder"
          />
        </p>
        <button
          class="link-button mx-3 fs-5 col-6 placeholder btn btn-light"
          disabled=""
          tabindex="-1"
          type="button"
        />
      </div>
    </div>
  </div>
  <div
    class="d-flex"
  >
    <div
      class="dark-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="placeholder-glow card-title h5"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <div
          class="mb-2 placeholder-glow card-subtitle h6"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <p
          class="placeholder-glow card-text"
        >
          <span
            class="col-6 placeholder"
          />
        </p>
        <button
          class="link-button mx-3 fs-5 col-6 placeholder btn btn-light"
          disabled=""
          tabindex="-1"
          type="button"
        />
      </div>
    </div>
    <div
      class="medium-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="placeholder-glow card-title h5"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <div
          class="mb-2 placeholder-glow card-subtitle h6"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <p
          class="placeholder-glow card-text"
        >
          <span
            class="col-6 placeholder"
          />
        </p>
        <button
          class="link-button mx-3 fs-5 col-6 placeholder btn btn-light"
          disabled=""
          tabindex="-1"
          type="button"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home page Use Jest snapshots to test the Home page renders Home page stocks table as expected 1`] = `
<DocumentFragment>
  <div
    class="d-flex justify-content-around align-items-center py-1 dark-bg position-sticky"
  >
    <a
      class="link-button mx-3 fs-5"
      href="/"
    />
    <h3
      class="body-title-text flex-grow-1 text-center mb-0 pt-1"
    >
      Stocks Watcher
    </h3>
    <button
      class="link-button fs-5"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 640 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M633.82 458.1l-157.8-121.96C488.61 312.13 496 285.01 496 256v-48c0-8.84-7.16-16-16-16h-16c-8.84 0-16 7.16-16 16v48c0 17.92-3.96 34.8-10.72 50.2l-26.55-20.52c3.1-9.4 5.28-19.22 5.28-29.67V96c0-53.02-42.98-96-96-96s-96 42.98-96 96v45.36L45.47 3.37C38.49-2.05 28.43-.8 23.01 6.18L3.37 31.45C-2.05 38.42-.8 48.47 6.18 53.9l588.36 454.73c6.98 5.43 17.03 4.17 22.46-2.81l19.64-25.27c5.41-6.97 4.16-17.02-2.82-22.45zM400 464h-56v-33.77c11.66-1.6 22.85-4.54 33.67-8.31l-50.11-38.73c-6.71.4-13.41.87-20.35.2-55.85-5.45-98.74-48.63-111.18-101.85L144 241.31v6.85c0 89.64 63.97 169.55 152 181.69V464h-56c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16z"
        />
      </svg>
    </button>
    <button
      class="link-button mx-3 fs-5"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 6a3.939 3.939 0 0 0-3.934 3.934h2C10.066 8.867 10.934 8 12 8s1.934.867 1.934 1.934c0 .598-.481 1.032-1.216 1.626a9.208 9.208 0 0 0-.691.599c-.998.997-1.027 2.056-1.027 2.174V15h2l-.001-.633c.001-.016.033-.386.441-.793.15-.15.339-.3.535-.458.779-.631 1.958-1.584 1.958-3.182A3.937 3.937 0 0 0 12 6zm-1 10h2v2h-2z"
        />
        <path
          d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
        />
      </svg>
    </button>
  </div>
  <form
    class="d-flex my-3 control-text position-sticky"
  >
    <input
      class="form-control form-control-sm"
      placeholder="Stock search..."
      type="text"
      value=""
    />
    <select
      class="form-select"
    >
      <option
        value="ALL"
      >
        All Exchanges
      </option>
      <option
        value="AMEX"
      >
        New York Stock Exchange Arca
      </option>
      <option
        value="NASDAQ"
      >
        Nasdaq Global Market
      </option>
      <option
        value="NYSE"
      >
        New York Stock Exchange
      </option>
    </select>
    <select
      class="form-select"
    >
      <option
        value="ALL"
      >
        All types
      </option>
      <option
        value="stock"
      >
        Stocks
      </option>
      <option
        value="etf"
      >
        ETFs
      </option>
      <option
        value="fund"
      >
        Funds
      </option>
    </select>
  </form>
  <ul
    class="d-flex justify-content-center pagination pagination-sm"
  >
    <li
      class="page-item active"
    >
      <span
        class="page-link"
      >
        1
        <span
          class="visually-hidden"
        >
          (current)
        </span>
      </span>
    </li>
  </ul>
  <div
    class="d-flex"
  >
    <div
      class="medium-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        >
          SPY - 435.18
        </div>
        <div
          class="mb-2 card-subtitle h6"
        >
          New York Stock Exchange Arca - AMEX
        </div>
        <p
          class="card-text"
        >
          SPDR S&P 500 ETF Trust - etf
        </p>
        <a
          class="link-button mx-3 fs-5"
          href="/details/SPY"
        >
          See details...
        </a>
      </div>
    </div>
    <div
      class="dark-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        >
          CMCSA - 52.69
        </div>
        <div
          class="mb-2 card-subtitle h6"
        >
          Nasdaq Global Select - NASDAQ
        </div>
        <p
          class="card-text"
        >
          Comcast Corporation - stock
        </p>
        <a
          class="link-button mx-3 fs-5"
          href="/details/CMCSA"
        >
          See details...
        </a>
      </div>
    </div>
  </div>
  <div
    class="d-flex"
  >
    <div
      class="dark-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        >
          KMI - 17.97
        </div>
        <div
          class="mb-2 card-subtitle h6"
        >
          New York Stock Exchange - NYSE
        </div>
        <p
          class="card-text"
        >
          Kinder Morgan, Inc. - stock
        </p>
        <a
          class="link-button mx-3 fs-5"
          href="/details/KMI"
        >
          See details...
        </a>
      </div>
    </div>
    <div
      class="medium-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        >
          INTC - 52.26
        </div>
        <div
          class="mb-2 card-subtitle h6"
        >
          Nasdaq Global Select - NASDAQ
        </div>
        <p
          class="card-text"
        >
          Intel Corporation - stock
        </p>
        <a
          class="link-button mx-3 fs-5"
          href="/details/INTC"
        >
          See details...
        </a>
      </div>
    </div>
  </div>
  <div
    class="d-flex"
  >
    <div
      class="medium-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        >
          MU - 66.38
        </div>
        <div
          class="mb-2 card-subtitle h6"
        >
          Nasdaq Global Select - NASDAQ
        </div>
        <p
          class="card-text"
        >
          Micron Technology, Inc. - stock
        </p>
        <a
          class="link-button mx-3 fs-5"
          href="/details/MU"
        >
          See details...
        </a>
      </div>
    </div>
    <div
      class="dark-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        >
          GDX - 32.15
        </div>
        <div
          class="mb-2 card-subtitle h6"
        >
          New York Stock Exchange Arca - AMEX
        </div>
        <p
          class="card-text"
        >
          VanEck Vectors Gold Miners ETF - etf
        </p>
        <a
          class="link-button mx-3 fs-5"
          href="/details/GDX"
        >
          See details...
        </a>
      </div>
    </div>
  </div>
  <div
    class="d-flex"
  >
    <div
      class="dark-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        >
          GE - 102.36
        </div>
        <div
          class="mb-2 card-subtitle h6"
        >
          New York Stock Exchange - NYSE
        </div>
        <p
          class="card-text"
        >
          General Electric Company - stock
        </p>
        <a
          class="link-button mx-3 fs-5"
          href="/details/GE"
        >
          See details...
        </a>
      </div>
    </div>
    <div
      class="medium-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        >
          BAC - 43.14
        </div>
        <div
          class="mb-2 card-subtitle h6"
        >
          New York Stock Exchange - NYSE
        </div>
        <p
          class="card-text"
        >
          Bank of America Corporation - stock
        </p>
        <a
          class="link-button mx-3 fs-5"
          href="/details/BAC"
        >
          See details...
        </a>
      </div>
    </div>
  </div>
  <div
    class="d-flex"
  >
    <div
      class="medium-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        >
          EEM - 51.16
        </div>
        <div
          class="mb-2 card-subtitle h6"
        >
          New York Stock Exchange Arca - AMEX
        </div>
        <p
          class="card-text"
        >
          iShares MSCI Emerging Markets ETF - etf
        </p>
        <a
          class="link-button mx-3 fs-5"
          href="/details/EEM"
        >
          See details...
        </a>
      </div>
    </div>
    <div
      class="dark-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        >
          XLF - 38.29
        </div>
        <div
          class="mb-2 card-subtitle h6"
        >
          New York Stock Exchange Arca - AMEX
        </div>
        <p
          class="card-text"
        >
          Financial Select Sector SPDR Fund - etf
        </p>
        <a
          class="link-button mx-3 fs-5"
          href="/details/XLF"
        >
          See details...
        </a>
      </div>
    </div>
  </div>
  <div
    class="d-flex"
  >
    <div
      class="dark-bg rounded-0 card"
      style="width: 50%;"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        >
          AAPL - 140.91
        </div>
        <div
          class="mb-2 card-subtitle h6"
        >
          Nasdaq Global Select - NASDAQ
        </div>
        <p
          class="card-text"
        >
          Apple Inc. - stock
        </p>
        <a
          class="link-button mx-3 fs-5"
          href="/details/AAPL"
        >
          See details...
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
